src/charm.py

```diff
@@ -672,6 +668,104 @@ def _get_cpu_extensions() -> list[str]:
                 del cpu_flags[0]
         return cpu_flags
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        """Get access interface NAD config.
+
+        Returns:
+            config (dict): Access interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if access_mtu := self._get_access_interface_mtu_config():
+            config.update({"mtu": access_mtu})
+        return config
+
+    def _get_core_nad_config(self) -> Dict[Any, Any]:
+        """Get core interface NAD config.
+
+        Returns:
+            config (dict): Core interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_core_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if core_mtu := self._get_core_interface_mtu_config():
+            config.update({"mtu": core_mtu})
+        return config
+
+    def _get_core_interface_mtu_config(self) -> Optional[str]:
+        """Get Core interface MTU size.
+
+        Returns:
+            mtu_size (str/None): If MTU size is not configured return None
+                                    If it is set, returns the configured value
+
+        """
+        return self.model.config.get("core-interface-mtu-size")
+
+    def _get_access_interface_mtu_config(self) -> Optional[str]:
+        """Get Access interface MTU size.
+
+        Returns:
+            mtu_size (str/None): If MTU size is not configured return None
+                                    If it is set, returns the configured value
+
+        """
+        return self.model.config.get("access-interface-mtu-size")
+
+    def _access_interface_mtu_size_is_valid(self) -> bool:
+        """Checks whether the access interface MTU size is valid.
+
+        Returns:
+            bool: Whether access interface MTU size is valid
+        """
+        if (access_mtu := self._get_access_interface_mtu_config()) is None:
+            return True
+        try:
+            return 1200 <= int(access_mtu) < 65537
```

Comment:
The IPv4 header Total Length is a 16 bit field.

```suggestion
            return 1200 <= int(access_mtu) <= 65535
```